great_expectations/execution_engine/execution_engine.py

```diff
@@ -307,7 +307,7 @@ def get_batch_data_and_markers(self, batch_spec) -> Tuple[BatchData, BatchMarker
     def resolve_metrics(  # noqa: C901 - 16
         self,
         metrics_to_resolve: Iterable[MetricConfiguration],
-        metrics: Optional[Dict[Tuple[str, str, str], MetricConfiguration]] = None,
+        metrics: Optional[Dict[Tuple[str, str, str], Any]] = None,
```

Comment:
I think we should keep `MetricConfiguration` if this the correct type.
What are the typing issues that made you use `Any`?